src/test/kotlin/org/jetbrains/kotlinx/jupyter/test/repl/ReplWithReceiverTests.kt

```diff
@@ -6,14 +6,41 @@ import org.jetbrains.kotlinx.jupyter.test.classpath
 import org.junit.jupiter.api.Test
 import java.io.File
 import kotlin.test.assertEquals
+import kotlin.test.assertFails
 
 class ReplWithReceiverTests : AbstractReplTest() {
     @Test
     fun testReplWithReceiver() {
         val value = 5
         val cp = classpath + File(ConstReceiver::class.java.protectionDomain.codeSource.location.toURI().path)
-        val repl = ReplForJupyterImpl(resolutionInfoProvider, cp, null, scriptReceivers = listOf(ConstReceiver(value)))
+        val repl = ReplForJupyterImpl(resolutionInfoProvider, cp, null, implicitReceivers = listOf(ConstReceiver(value)))
         val res = repl.eval("value")
         assertEquals(value, res.resultValue)
     }
+
+    @Test
+    fun testReplWithAdHocReceivers() {
+        val repl = makeSimpleRepl()
+        repl.eval(
+            """
+            class A(val x: Int)
+            object B { val y = 42 }
+            """.trimIndent()
+        )
+
+        repl.eval(
+            """
+            HOST.withReceiver(A(50))
```

Comment:
Still look somewhat strange. Not everybody understands what `host` is.

Reply:
Maybe `executor`?

Reply:
From the logical point of view, the receiver is applied to cells. So the code in cells is wrapped in the receiver. This is why I initially proposed to attach it to the notebook. It seems like a non-important thing, but discoverability is important.